feat: Add OpenAI-compliant tool calls visibility to chat completions

wtfsayo opened the pull request:
## Summary

This PR adds support for viewing intermediate tool calls and results in the chat completions API while maintaining full OpenAI API compliance.

## Changes

- **OpenAI API Compliance**: Default responses remain fully compliant with OpenAI's chat completions API
- **Optional Tool Call Visibility**: Add `include_intermediate_steps` parameter to expose tool execution flow
- **Proper Tool Call Formatting**: Tool calls follow OpenAI's exact format with `tool_calls` array
- **Tool Result Messages**: Tool results are returned as proper `tool` role messages with `tool_call_id`
- **Multi-step Execution**: Support for agents with multiple tool execution steps

## Usage

### Standard OpenAI Response (default)
```bash
curl -X POST http://localhost:3000/v1/chat/completions \
  -H 'Content-Type: application/json' \
  -d '{"agent_id":"default","messages":[{"role":"user","content":"What is my wallet address?"}]}'
```

Returns standard OpenAI format:
```json
{
  "choices": [{
    "message": {
      "role": "assistant",
      "content": "Your wallet address is: 0x123...",
      "tool_calls": [...] // Only if tools were used
    },
    "finish_reason": "stop" // or "tool_calls"
  }]
}
```

### With Tool Call Visibility
```bash
curl -X POST http://localhost:3000/v1/chat/completions \
  -H 'Content-Type: application/json' \
  -d '{"agent_id":"default","messages":[{"role":"user","content":"What is my wallet address?"}],"include_intermediate_steps":true}'
```

Returns additional `conversation_flow` with complete execution trace:
```json
{
  "choices": [...],
  "conversation_flow": [
    {"role": "user", "content": "What is my wallet address?"},
    {"role": "assistant", "tool_calls": [{"id": "call_123", "function": {"name": "getWalletAddress"}}]},
    {"role": "tool", "tool_call_id": "call_123", "content": "{\"address\":\"0x123...\"}"},
    {"role": "assistant", "content": "Your wallet address is: 0x123..."}
  ]
}
```

## Benefits

- **Full OpenAI Compatibility**: Works with any OpenAI-compatible client
- **Complete Traceability**: See exactly what tools were called and their results
- **Debugging Support**: Understand agent decision-making process
- **Backward Compatible**: No breaking changes to existing API usage

wtfsayo committed:
feat: Add OpenAI-compliant tool calls visibility to chat completions

- Add optional include_intermediate_steps parameter to chat completions API
- Return conversation_flow with detailed tool call execution when requested
- Maintain full OpenAI API compliance by default
- Include tool_calls in assistant messages when tools are used
- Format tool results as proper tool role messages
- Support multi-step agent execution with complete traceability

This enables debugging and monitoring of agent tool usage while maintaining
compatibility with OpenAI API clients.

ChristopherTrimboli commented on src/server.ts:
```diff
@@ -108,8 +109,97 @@ app.post("/v1/chat/completions", async (c: Context) => {
+            role: "assistant",
+            content: result.text,
+          },
+          finish_reason: "stop",
```
@wtfsayo this looks valid, need to support regular calls no tools too

wtfsayo committed:
Fix user message selection in chat completions

Updates the logic to correctly find the last user message in the conversation when including intermediate steps. This ensures the original user message is accurately added to the conversation messages.
